components/playback/controls/player/AddCustomer.vue

```diff
@@ -25,54 +63,91 @@ import BootstrapVue from "bootstrap-vue";
+const TextValidation_Alphanumeric = new TextValidation("alphanumeric");
+const TextValidation_Nickname = new TextValidation("nickname");
 export default {
   name: "AddCustomer",
   components: {
     InputWidget,
-    // ButtonWidget,
+    ButtonWidget,
   },
   props: {
     dialogdata: { type: Object, default: null },
     dataindex: { type: Number, default: 0 },
   },
   data() {
     return {
+      uuid: "",
+      apikey: "",
+      nickname: "",
       error_text_uuid: "This field is required",
+      error_text_api: "",
+      error_text_nickname: "This field is required",
+      enablelist_add_customer: [true, false],
     };
   },
-  created: function () {
-    this.label_uuid = "Enter Alphanumeric ID";
-    this.label_api = "Enter API Key(Optional)";
-    this.label_nickname = "Enter ID Nickname";
-    this.keyplaceholder_encode = "2VSckkBYr2An3dqHEyfRRE";
-    this.keyplaceholder_uuid = "ba86b8f0-6fdf-4944-87a0-8a491a19490e";
-    this.keyplaceholder_nickname = "Curi Bio Main Account";
-    this.spellchecking = false;
-
-    this.error_text_api = "";
-    this.error_text_nickname = "This field is required";
-    this.key_validation = false;
-    this.entry_width = 400;
-    this.disallow_entry = false;
-    this.btnnames_add_customer = ["Cancel", "Save ID"];
-    this.enablelist_add_customer = [true, false];
-    this.visiblecolor_add_customer = "#FFFFFF";
-    this.hidecolor_add_customer = "#3F3F3F";
-    this.hovercolors_add_customer = ["#BD4932", "#19ac8a"];
-  },
   methods: {
     on_update_uuid: function (new_value) {
       this.error_text_uuid = TextValidation_UUIDBase57.validate(new_value);
+      this.uuid = new_value;
+      this.enable_save_button();
+    },
+    on_update_api: function (new_value) {
+      this.error_text_api = TextValidation_Alphanumeric.validate(new_value);
+      this.apikey = new_value;
+      this.enable_save_button();
+    },
+    on_update_nickname: function (new_value) {
+      this.error_text_nickname = TextValidation_Nickname.validate(new_value);
+      this.nickname = new_value;
+      this.enable_save_button();
+    },
+    clicked_button: function (choice) {
+      switch (choice) {
+        case 0:
+          this.cancel_addcustomer();
+          break;
+        case 1:
+          this.save_newcustomer();
+          break;
+      }
+    },
+    cancel_addcustomer() {
+      this.$bvModal.hide("add-customer");
+    },
+    save_newcustomer() {
+      const add_customer = {
+        cust_id: this.dataindex,
+        uuid: this.uuid,
+        api_key: this.apikey,
+        nickname: this.nickname,
+        user_ids: [],
+      };
+      this.$emit("save-id", add_customer);
+      this.$bvModal.hide("add-customer");
+    },
+    enable_save_button() {
+      if (this.error_text_uuid === "") {
+        if (this.error_text_api === "") {
+          if (this.error_text_nickname === "") {
+            this.enablelist_add_customer = [true, true];
```

Comment:
you can add a `return` statement after setting `[true, true]`,  and then just have one line of code setting all other cases to `[true, false]`.  This will help deduplicate the code within this function

Reply:
Completed

Reply:
the following is what I meant to reduce the number of duplicated statements

`if (this.error_text_uuid === "") {`

`    if (this.error_text_api === "") {`

`          if (this.error_text_nickname === "") {`

`            this.enablelist_add_customer = [true, true];`

`           return`

`          }`

`     }`

`}`

`this.enablelist_add_customer = [true, false];`